plugins/net_plugin/net_plugin.cpp

```diff
@@ -3004,31 +2983,33 @@ namespace eosio {
                      auto check_time = check->last_handshake_sent.time + check->last_handshake_recv.time;
                      g_check_conn.unlock();
                      if (msg.time + c_time <= check_time)
-                        continue;
+                        return false;
                   } else if (net_version < proto_dup_node_id_goaway || msg.network_version < proto_dup_node_id_goaway) {
                      if (my_impl->p2p_address < msg.p2p_address) {
                         fc_dlog( logger, "my_impl->p2p_address '${lhs}' < msg.p2p_address '${rhs}'",
                                  ("lhs", my_impl->p2p_address)( "rhs", msg.p2p_address ) );
                         // only the connection from lower p2p_address to higher p2p_address will be considered as a duplicate,
                         // so there is no chance for both connections to be closed
-                        continue;
+                        return false;
                      }
                   } else if (my_impl->node_id < msg.node_id) {
                      fc_dlog( logger, "not duplicate, my_impl->node_id '${lhs}' < msg.node_id '${rhs}'",
                               ("lhs", my_impl->node_id)("rhs", msg.node_id) );
                      // only the connection from lower node_id to higher node_id will be considered as a duplicate,
                      // so there is no chance for both connections to be closed
-                     continue;
+                     return false;
                   }
-
-                  g_cnts.unlock();
-                  peer_dlog( this, "sending go_away duplicate, msg.p2p_address: ${add}", ("add", msg.p2p_address) );
-                  go_away_message gam(duplicate);
-                  gam.node_id = conn_node_id;
-                  enqueue(gam);
-                  no_retry = duplicate;
-                  return;
+                  return true;
                }
+               return false;
+            };
+            if (my_impl->connections.any_of_connections(std::move(is_duplicate))) {
```

Comment:
looks good, but as a matter of preference I'd call the function simply `any_of` because it is always used as:

`connections.any_of_connections`, and I don't think the repetition adds anything.

Reply:
Left it since I added an `any_of_block_connections`

Reply:
Cool, but I think your `any_of_block_connections` is missing the `block` check.

Reply:
Thanks. Fixed.